src/components/engagement/handlers/set-initial-end-date.handler.ts

```diff
@@ -52,10 +63,15 @@ export class SetInitialEndDate implements IEventHandler<SubscribedEvent> {
     try {
       const initialEndDate = engagement.endDate.value;
 
-      const updatedEngagement = await this.updateEngagementInitialEndDate(
+      await this.updateEngagementInitialEndDate(
         engagement,
         initialEndDate,
-        event.session
+        engagement.changeset
+      );
+      const updatedEngagement = await this.engagementService.readOne(
+        engagement.id,
+        event.session,
+        viewOfChangeset(engagement.changeset)
       );
```

Comment:
Can't this read call be replaced by just setting the property directly?
```suggestion
      const updatedEngagement = {
        ...engagement,
        initialEndDate: {
           ...engagement.initialEndDate,
           value: initialEndDate,
        },
      };
```